File: contextflow-core/sdk/python/contextflow/exporters/claude.py
"""Claude exporter"""

from typing import Optional
from .base import BaseExporter, ExporterConfig
from ..types import ContextFlowFile


class ClaudeExporter(BaseExporter):
    """Export ContextFlow to Claude system prompt (Markdown)"""

    @property
    def name(self) -> str:
        return "claude"

    @property
    def output_format(self) -> str:
        return "text/markdown"

    def export(self, artifact: ContextFlowFile, config: Optional[ExporterConfig] = None) -> str:
        """Export ContextFlow to Claude-compatible Markdown system prompt"""
        config = config or ExporterConfig()

        sections = []

        # Add preferences
        if config.include_preferences and artifact.preferences:
            sections.append(self._format_preferences(artifact.preferences))

        # Add notes/knowledge
        if config.include_notes and artifact.notes:
            sections.append(self._format_notes(artifact.notes))

        # Add recent conversations
        if config.include_conversations and artifact.conversations:
            conversations = artifact.conversations
            if config.max_conversations:
                conversations = conversations[-config.max_conversations :]

            sections.append(self._format_conversations(conversations))

        return "\n\n".join(sections)

    def _format_preferences(self, preferences) -> str:
        """Format preferences as Markdown"""
        lines = ["# User Preferences"]

        if preferences.languages:
            lines.append(f"- **Languages**: {', '.join(preferences.languages)}")

        if preferences.frameworks:
            lines.append(f"- **Frameworks**: {', '.join(preferences.frameworks)}")

        if preferences.style:
            lines.append(f"- **Style**: {preferences.style}")

        if preferences.tone:
            lines.append(f"- **Tone**: {preferences.tone}")

        return "\n".join(lines)

    def _format_notes(self, notes) -> str:
        """Format notes as Markdown"""
        lines = ["# Knowledge Base"]

        for note in notes:
            lines.append(f"\n## {note.title or 'Untitled'}")
            lines.append(note.content)

        return "\n".join(lines)

    def _format_conversations(self, conversations) -> str:
        """Format conversations as Markdown"""
        lines = ["# Recent Conversations"]

        for conv in conversations:
            lines.append(f"\n## {conv.title or 'Untitled Conversation'}")
            if conv.created:
                lines.append(f"*Date: {conv.created}*")

            for msg in conv.messages:
                role_label = msg.role.capitalize()
                lines.append(f"\n**{role_label}**: {msg.content}")

        return "\n".join(lines)

File: contextflow-core/sdk/javascript/src/exporters/claude.ts
import { BaseExporter, ExporterConfig } from './base';
import { ContextFlowFile } from '../types';

export class ClaudeExporter extends BaseExporter {
  readonly name = 'claude';
  readonly targetPlatform = 'anthropic';

  async export(
    contextflow: ContextFlowFile,
    config: ExporterConfig = {}
  ): Promise<string> {
    const parts: string[] = [];

    // Add preferences
    if (contextflow.preferences) {
      parts.push('# User Preferences');

      if (contextflow.preferences.languages) {
        parts.push(`Languages: ${contextflow.preferences.languages.join(', ')}`);
      }
      if (contextflow.preferences.frameworks) {
        parts.push(`Frameworks: ${contextflow.preferences.frameworks.join(', ')}`);
      }
      if (contextflow.preferences.style) {
        parts.push(`Style: ${contextflow.preferences.style}`);
      }
      if (contextflow.preferences.tone) {
        parts.push(`Tone: ${contextflow.preferences.tone}`);
      }
      parts.push('');
    }

    // Add notes
    if (contextflow.notes && contextflow.notes.length > 0) {
      for (const note of contextflow.notes) {
        parts.push(`## ${note.title || 'Note'}`);
        parts.push(note.content);
        parts.push('');
      }
    }

    // Optionally add recent conversation context
    if (contextflow.conversations && contextflow.conversations.length > 0) {
      const recentConvos = contextflow.conversations.slice(-3); // Last 3
      parts.push('# Recent Context');
      for (const conv of recentConvos) {
        parts.push(`- ${conv.title || 'Untitled Conversation'}`);
      }
      parts.push('');
    }

    return parts.join('\n');
  }
}
